### PR TITLE
🕹 Feat/add stories

### PR DESCRIPTION
**Pull request by rayrutjes**
https://59e661aca6188f358de1cf25--vue-instantsearch.netlify.com/stories/

cc @mthuret @iam4x @samouss

### CODE REVIEW

File: docs/src/stylesheets/components/_buttons.sass [hunk 1]
```diff
@@ -80,7 +80,7 @@
 
 .btn-static-theme
   border: none
-  color: #fff
+  color: #fff !important
```
Comment (Haroenv) [inline docs/src/stylesheets/components/_buttons.sass]: I'm not sure if we should add this

File: src/components/Results.vue [hunk 2]
```diff
@@ -3,7 +3,7 @@
 
     <slot name="header"></slot>
 
-    <slot v-for="result in results" :result="result" :key="result.objectID">
+    <slot v-for="result in results" :result="result">
```
Comment (Haroenv) [inline src/components/Results.vue]: Why was this removed? Now images will be replaced weirdly, right?

**Comment by samouss** [comment]
Nice! I had planned to do it too :)

File: src/components/Results.vue (hunk 2 above)
Comment (samouss) [reply src/components/Results.vue]: The `slot` can't have the `:key` attribute (an error is raised) but the `template` can (see [this issue](https://github.com/vuejs/vue/issues/5575)).

File: docs/src/stylesheets/components/_buttons.sass (hunk 1 above)
Comment (rayrutjes) [reply docs/src/stylesheets/components/_buttons.sass]: Dirty fix 🙈

File: src/components/Results.vue (hunk 2 above)
Comment (rayrutjes) [reply src/components/Results.vue]: > Why was this removed? Now images will be replaced weirdly, right?

The key wasn't used so far. We should try to use a key on every hit to see if that fixes the images issue.

**Comment by rayrutjes** [comment]
> is it possible to pass some arguments to it? To be able to simulate a search without results or some others use cases

I guess everything is possible ;) I kept it simple for now but feel free to implement whatever you think makes sense!

**Comment by rayrutjes** [comment]
I'll merge this so we can iterate, some good ideas here  🚀  feel free to implement stuff!

**Comment by mthuret** [comment]
Awesome @rayrutjes o/

**Comment by rayrutjes** [comment]
We will need your expertise @mthuret on StoryBook to move it forward!
Super cool stuff, it was seamless to integrate.

**Comment by mthuret** [comment]
For sure, some things I can see right away: 

- we could share the same look and feel -> ie the frame around every widgets
- I usually add a `playground` story if there are some props the user can play with. 
- about each stories to create, I mostly tried to add variation of the widget state that are important for a user. 

@marielaures is currently working on adding some addons to the react storybook (usage & accessibility), she might be helpful working on this one too.

**Comment by rayrutjes** [comment]
That sounds great @mthuret .

> we could share the same look and feel -> ie the frame around every widgets

Yes yes yes 👏 , I didn't do any styling and eventually was hoping you would say we can share the styles 🚀 

> I usually add a playground story if there are some props the user can play with

I did see the playground part in react-instantsearch stories, but didn't quite understand everything. Sounds super kool though.

> about each stories to create, I mostly tried to add variation of the widget state that are important for a user.

I added most of the default implementations, and now we can iterate on adding more. I'd love having someone from the team working on it!
